pkg/asset/installconfig/powervs/session.go

```diff
@@ -42,35 +48,67 @@ type Session struct {
      4) put it into Platform {userid: , iamtoken: , ...}
 */
 func GetSession() (*Session, error) {
-	s, err := getPISession()
+	s, uc, err := getPISession()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to load credentials")
 	}
-
-	return &Session{Session: s}, nil
+	return &Session{Session: s, Creds: uc}, nil
 }
 
 /*
 //  https://github.com/IBM-Cloud/power-go-client/blob/master/ibmpisession/ibmpowersession.go
 */
-func getPISession() (*ibmpisession.IBMPISession, error) {
+func getPISession() (*ibmpisession.IBMPISession, *UserCredentials, error) {
 
 	var (
-		id, passwd, region, zone string
+		id, passwd, apikey, region, zone string
 	)
 
 	if id = os.Getenv("IBMID"); len(id) == 0 {
-		return nil, errors.New("empty IBMID environment variable")
+		err := survey.Ask([]*survey.Question{
+			{
+				Prompt: &survey.Input{
+					Message: "IBM Cloud User ID",
+					Help:    "The login for \nhttps://cloud.ibm.com/",
+				},
+			},
+		}, &id)
+		if err != nil {
+			return nil, nil, errors.New("empty IBMID environment variable")
+		}
 	}
 	if passwd = os.Getenv("IBMID_PASSWORD"); len(passwd) == 0 {
-		return nil, errors.New("empty IBMID_PASSWORD variable")
-	}
-
+		err := survey.Ask([]*survey.Question{
+                        {
+                                Prompt: &survey.Password{
+                                        Message: "IBM Cloud User Password",
+                                        Help:    "The login for \nhttps://cloud.ibm.com/",
+                                },
+                        },
+                }, &passwd)
+                if err != nil {
+                        return nil, nil, errors.New("empty IBMID_PASSWORD environment variable")
+                }
+        }
+        if apikey = os.Getenv("API_KEY"); len(apikey) == 0 {
+                err := survey.Ask([]*survey.Question{
+                        {
+                                Prompt: &survey.Password{
+					Message: "IBM Cloud API Key",
+					Help:    "The api key installation.\nhttps://cloud.ibm.com/iam/apikeys",
+                                },
+                        },
+                }, &apikey)
+                if err != nil {
+                        return nil, nil, errors.New("empty API_KEY environment variable")
```

Comment:
wouldn't this error happen because something went wrong with the survey question? can you change the error returned to something like, "Error saving the API key"

Reply:
That holds true for all three of these, I can change it to whatever you want.  However if something went wrong with the survey it seems to return an empty string, for example if I ctrl c out of it.

Reply:
got it. to me, this message seemed to tell the user that they'd not set a local api key var. and that may or may not have been the case it sounds like.